Decred Wallet/Custom Views/FloatingLabelTextInput.swift

```diff
@@ -0,0 +1,123 @@
+    let floatingLabelBottomPaddingNormal:CGFloat = 33
+    let floatingLabelBottomPaddingActive:CGFloat = 8
+    let floatingLabelColorNormal:UIColor = UIColor.appColors.bluishGray
+    let floatingLabelColorActive:UIColor = UIColor.appColors.decredBlue
+    var isInErrorState:Bool = false
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.initView()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.initView()
+    }
+    
+    private func initView() {
+        border.cornerRadius = self.cornerRadius
+        border.borderColor = self.inactiveBorderColor.cgColor
+        border.borderWidth = self.borderWidth
+        self.layer.addSublayer(border)
+        
+        self.initFloatedLabel()
+        self.addTarget(self, action: #selector(self.showFloatingLabel), for: .editingDidBegin)
+        self.addTarget(self, action: #selector(self.hideFloatingLabel), for: .editingDidEnd)
+    }
+    
+    private func initFloatedLabel() {
+        floatingLabel.backgroundColor = UIColor.white
+        floatingLabel.textColor = self.floatingLabelColorNormal
+        floatingLabel.font = UIFont(name: self.font!.familyName, size: 16)
+        floatingLabel.clipsToBounds = true
+        floatingLabel.translatesAutoresizingMaskIntoConstraints = false
+        floatingLabel.sizeToFit()
+        floatingLabel.leftInset = 4
+        floatingLabel.rightInset = 4
+        self.floatingLabel.text = self.placeholder
+        self.placeholder = ""
+        self.addSubview(self.floatingLabel)
+        self.floatingLabelBottomConstraint = self.floatingLabel.bottomAnchor.constraint(equalTo: self.topAnchor, constant: self.floatingLabelBottomPaddingNormal)
+        NSLayoutConstraint.activate([
+            self.floatingLabel.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 8),
+            floatingLabelBottomConstraint!
+        ])
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        border.frame = self.bounds
+    }
+    
+    func showError() {
+        self.isInErrorState = true
+        showFloatingLabel()
+    }
+    
+    func hideError() {
+        self.isInErrorState = false
+        showFloatingLabel()
+    }
+    
+    @objc func showFloatingLabel() {
+        border.borderColor = self.isInErrorState ? UIColor.appColors.decredOrange.cgColor : UIColor.appColors.decredBlue.cgColor
+        UIView.animate(withDuration: 0.13) {
+            self.floatingLabelBottomConstraint?.constant = self.floatingLabelBottomPaddingActive
+            self.floatingLabel.textColor = self.isInErrorState ? UIColor.appColors.decredOrange : self.floatingLabelColorActive
+            self.floatingLabel.font = UIFont(name: self.font!.familyName, size: 14)
+            self.layoutIfNeeded()
+        }
+    }
+    
+    @objc func hideFloatingLabel() {
+        border.borderColor = self.inactiveBorderColor.cgColor
```

Comment:
this `hideFloatingLabel` function should not be responsible for updating the border color, if it must be responsible, rename the function to `hideFloatingLabelAndUpdateBorderColor`